### ISSUE TITLE
dmd 2.069.1 regression, std.concurrency functions sleep forever

### ISSUE THREAD

**Issue by NCrashed** [rejectedsoftware/vibe.d#1343]
I move my old project from dmd 2.066 to latest release. The problem: `std.concurrency` functions (
`receive`, `receiveOnly`, `receiveTimeout`) hang vibed's fibers.

Demo of issue
===========

dub.sdl
```
name "threads-test"
dependency "vibe-d" version="==0.7.27-alpha.1"
versions "VibeDefaultMain"
```

app.d
``` D
import std.stdio;
import std.concurrency : 
    stdReceiveOnly = receiveOnly
  , stdThisTid = thisTid
  , StdTid = Tid
  , stdSend = send
  , spawn;
import vibe.d;

void thread1(StdTid mainThread, StdTid otherThread) {

  writeln("thread1 send");
  otherThread.stdSend(stdThisTid);
  writeln("thread1 wait");
  stdReceiveOnly!StdTid();

  writeln("thread1 exit");
  mainThread.stdSend(true);
}

void thread2(StdTid mainThread) {
  writeln("thread2 wait");
  std.concurrency.receive((StdTid otherThread) {
      writeln("thread2 send");
      otherThread.stdSend(stdThisTid);
    });

  writeln("thread2 exit");
  mainThread.stdSend(true);
}

void test()
{
  auto tid2 = spawn(&thread2, stdThisTid);
	auto tid1 = spawn(&thread1, stdThisTid, tid2);

  stdReceiveOnly!bool();
  stdReceiveOnly!bool();
  writeln("test exit");
}

shared static this()
{
  auto settings = new HTTPServerSettings;
  settings.port = 8080;

  listenHTTP(settings, &handleRequest);
}

void handleRequest(HTTPServerRequest req,
                   HTTPServerResponse res)
{
  test();
  if (req.path == "/")
    res.writeBody("Hello, World!", "text/plain");
}
```

**Vibe-d**: 0.7.27-alpha.1
**dmd**: 2.069.1
**platform**: Fedora 23 x86_64

P.S. Dirty workaround is to run your stuff in separate thread and communicate with a fiber via shared memory.

**Comment by s-ludwig** [comment]
Fixed by d8b51c2f00a6d91042869de5d6e16d6380cad861 - `std.concurrency` integration simply got disabled a while ago, so that it used the default implementation, which blocks the event loop.

**Comment by s-ludwig** [comment]
The fixed caused troubles and requires a more involved fix.
